### PR TITLE
Add eo:snow_cover definition

### LINKED ISSUE #8: Support eo:snow_cover as attribute
EO processors often detect snow cover percentages in addition to cloud cover. Currently these information needs to be stored in custom properties which makes them not systematically discoverable. Therefore it would be good if this can be supported by the extension as an optional property, e.g. `eo:snow_cover`.

### PR DESCRIPTION
**Pull request by torse**
Initial minimalistic PR for eo:snow_cover field

### CODE REVIEW

**Comment by cholmes** [comment]
This looks good, though I'm wondering about just snow, vs snow and ice together? Or is snow meant to also imply ice? At Planet in our UDM we do `snow_ice_percent`: Percent of snow and ice values in dataset. Snow_ice values represent scene content areas (non-blackfilled) that are hidden below snow and/or ice.

If yours does cover ice too then probably good to have it in the description. 

I am curious on the datasets you're thinking of that use this. Just want to be sure it cuts across a few, and fits with all of them.

**Comment by torse** [comment]
Good point! I double checked with our data provider and indeed the attribute also imply ice covered surfaces. I've updated the description in the README.md accordingly but left the schema.json as is. Let me know you think this should be updated as well (e.g. `"title": "Snow/Ice Cover"`).

Our datasets are Sentinel-2 based [L2](https://doi.org/10.15489/ifczsszkcp63) and [L3](https://doi.org/10.15489/4hcq6dgkj648) products over Germany computed by the MAJA and WASP processor respectively. The access to the datasets via STAC are currently beeing developed and available for preview ([S2_L2A_MAJA](https://geoservice.dlr.de/eoc/ogc/stac/collections/S2_L2A_MAJA) and [S2_L3A_WASP](https://geoservice.dlr.de/eoc/ogc/stac/collections/S2_L3A_WASP)). In an technology project we've also provided experimental access to the original Sentinel-2 data as COGs that also include the S2 specific `snow_ice_percentage` attribute: [S2_L2A_MSI_COG](https://geoserver.dlr.loose.eox.at/ogc/stac/collections/S2_L2A_MSI_COG).

**Review by emmanuelmathot — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -58,6 +58,11 @@ Estimate of cloud cover as a percentage (0-100) of the entire scene. If not avai
 this value should be used in Item Properties rather than Item Assets, as an Item from an electro-optical source is a single snapshot 
 of the Earth, so the cloud cover value would apply to all assets. 
 
+#### eo:snow_cover
+
+Estimate of snow or ice covered surfaces as a percentage (0-100) of the entire scene. If not available, the field should not be provided. Generally, 
+this value should be used in Item Properties rather than Item Assets, as an Item from an electro-optical source is a single snapshot 
+of the Earth, so the snow cover value would apply to all assets.
```
Comment (m-mohr) [inline README.md:65]:
```suggestion
of the Earth, so the snow cover value would apply to all assets. 

```

File: README.md [hunk 2]
```diff
@@ -58,6 +58,11 @@ Estimate of cloud cover as a percentage (0-100) of the entire scene. If not avai
 this value should be used in Item Properties rather than Item Assets, as an Item from an electro-optical source is a single snapshot 
 of the Earth, so the cloud cover value would apply to all assets. 
 
+#### eo:snow_cover
+
+Estimate of snow or ice covered surfaces as a percentage (0-100) of the entire scene. If not available, the field should not be provided. Generally,
```
Comment (m-mohr) [inline README.md]:
```suggestion
Estimate of snow and ice covered surfaces as a percentage (0-100) of the entire scene. If not available, the field should not be provided. Generally, 
```

File: json-schema/schema.json [hunk 3]
```diff
@@ -126,6 +126,12 @@
           "minimum": 0,
           "maximum": 100
         },
+        "eo:snow_cover": {
+          "title": "Snow Cover",
```
Comment (m-mohr) [inline json-schema/schema.json]:
```suggestion
          "title": "Snow and Ice Cover",
```

**Review by m-mohr — APPROVED** [review]

**Comment by pjhartzell** [comment]
I'm late to the discussion here, but wondering why snow_cover is being added? Is there something particular about this ground cover type (as opposed to say, land, ocean, vegetated, barren, etc.) that impacts the electro-optical observation data? Sensor saturation?

**Pull request merged**